pytorch_lightning/core/hooks.py

```diff
@@ -385,10 +385,12 @@ def prepare_data(self):
 
     def train_dataloader(self) -> DataLoader:
```

Comment:
```suggestion
    def train_dataloader(self) -> Any:
```

Reply:
why Any?

Reply:
that's what the PR added here: https://github.com/PyTorchLightning/pytorch-lightning/pull/6076/files#diff-667a2513c158c2b01d73b479e4dea75587d8531f9cd286891d34570a2fd145dcR428

since `(list, dict, nested lists and dicts)` are supported for train_dataloader. although Any is too over simplified but we can be more specific here.

Reply:
Do we document anywhere all those options? We should be clearer about it